rust/toolchain.bzl

```diff
@@ -296,6 +297,11 @@ rust_toolchain = rule(
             ),
             mandatory = True,
         ),
+        "target_json": attr.label(
```

Comment:
Is it possible you're testing this code locally without sandboxing? I don't see where the target_json file is passed to the rustc compile action, I only see the path to the file used in command line arguments. We definitely need to pass the file as an input to an action, otherwise remote execution and sandboxing won't work.

Reply:
I agree, I'll pass the file as a dep, probably during `collect_inputs` as a `compile_input`. I think the reason this was working is I am using a very kludgy script to package the json and use it during build time in order to work around an interesting behavior in rustc.

If you pass a custom target specification then rustc will append a hash of the absolute path to the JSON, turning something like `x86_64-unknown-none` into `x86_64-unknown-none-9187537878920592665`. This is done to differentiate specs with the same name but different paths. It also has the effect of making it impossible to compile with a prebuilt set of standard libraries. At least for me, my remote workers use temporary, nondetermistic working directories and so I have been copying my specification to `/tmp` using my kludge script.